presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeCtasQueries.java

```diff
@@ -0,0 +1,152 @@
+                "SELECT count(*) FROM orders");
+
+        assertCreateTableAsSelect(
+                "test_with_no_data",
+                "SELECT * FROM orders WITH NO DATA",
+                "SELECT * FROM orders LIMIT 0",
+                "SELECT 0");
+
+        // Tests for CREATE TABLE with UNION ALL: exercises PushTableWriteThroughUnion optimizer
+
+        assertCreateTableAsSelect(
+                "test_union_all",
+                "SELECT orderdate, orderkey, totalprice FROM orders WHERE orderkey % 2 = 0 UNION ALL " +
+                        "SELECT orderdate, orderkey, totalprice FROM orders WHERE orderkey % 2 = 1",
+                "SELECT orderdate, orderkey, totalprice FROM orders",
+                "SELECT count(*) FROM orders");
+
+        assertCreateTableAsSelect(
+                Session.builder(getSession()).setSystemProperty("redistribute_writes", "true").build(),
+                "test_union_all",
+                "SELECT CAST(orderdate AS DATE) orderdate, orderkey, totalprice FROM orders UNION ALL " +
+                        "SELECT DATE '2000-01-01', 1234567890, 1.23",
+                "SELECT orderdate, orderkey, totalprice FROM orders UNION ALL " +
+                        "SELECT DATE '2000-01-01', 1234567890, 1.23",
+                "SELECT count(*) + 1 FROM orders");
+
+        assertCreateTableAsSelect(
+                Session.builder(getSession()).setSystemProperty("redistribute_writes", "false").build(),
+                "test_union_all",
+                "SELECT CAST(orderdate AS DATE) orderdate, orderkey, totalprice FROM orders UNION ALL " +
+                        "SELECT DATE '2000-01-01', 1234567890, 1.23",
+                "SELECT orderdate, orderkey, totalprice FROM orders UNION ALL " +
+                        "SELECT DATE '2000-01-01', 1234567890, 1.23",
+                "SELECT count(*) + 1 FROM orders");
+
+        assertExplainAnalyze();
+        assertQuery("SELECT * from analyze_test", "SELECT orderstatus FROM orders");
+        assertUpdate("DROP TABLE analyze_test");
+
+    }
+    protected void assertCreateTableAsSelect(String table, @Language("SQL") String query, @Language("SQL") String rowCountQuery)
+    {
+        assertCreateTableAsSelect(getSession(), table, query, query, rowCountQuery);
+    }
+
+    protected void assertCreateTableAsSelect(String table, @Language("SQL") String query, @Language("SQL") String expectedQuery, @Language("SQL") String rowCountQuery)
+    {
+        assertCreateTableAsSelect(getSession(), table, query, expectedQuery, rowCountQuery);
+    }
+
+    protected void assertCreateTableAsSelect(Session session, String table, @Language("SQL") String query, @Language("SQL") String expectedQuery, @Language("SQL") String rowCountQuery) {
+        assertUpdate(session, "CREATE TABLE " + table + " AS " + query, rowCountQuery);
+        assertQuery(session, "SELECT * FROM " + table, expectedQuery);
+        assertUpdate(session, "DROP TABLE " + table);
```

Comment:
Do you need to test DROP TABLE in the CTAS test?
You would need to allow DROP TABLE in https://github.com/prestodb/presto/blob/dd5b168ff5bdf306de70be19cc37c9797ca00981/presto-native-execution/src/test/java/com/facebook/presto/nativeworker/NativeQueryRunnerUtils.java

and add `hive.allow-drop-table=true` as a hive property so you can drop the table - after you found that this is missing.

Reply:
Thanks @czentgr for the comments. I did exactly that and it works , but these tests will only work locally/on my branch unless `hive.allow-drop-table=true` is being set in the main codebase. Should I add the property `hive.allow-drop-table=true` in the codebase and try to raise a different PR mentioning the access issues?

Reply:
That's why I'm asking if you truly need to test DROP table here. After all, this is a pure catalog operation which certainly is tested and possibly doesn;t need to be done for the native worker as it is unrelated.

Reply:
It's not needed per say. Part of the reason why I was using them was because I don't want many tables to get created and persist there. I still do not know whether the tables are deleted after the tests are ran or they persist. If they are deleted, DROP operation is not needed.

Reply:
Yes, normally a good though to clean up. In this case, afaik, it is not needed because they won't persist past the test run in the hive metastore. Note how new tables are created at the beginning of the test as well.

Reply:
Okay, understood.

Reply:
I think we should stick with the DROP operation always. I agree with @pdabre12 that cleaning up is ideal. Enabling the API to not delete the tables can lead to bugs.
I don't see any harm in adding `hive.allow-drop-table=true` to the Native runner. I assume the Java runner has this as well.